File: src/types/review-cockpit.ts
import type { AgentPromptEntry } from './terminal';
import type { WorkspaceChangedFile, WorkspaceFileDiff } from './git-review';
import type { WorkspaceMergeReadiness } from './merge-readiness';
import type { WorkspaceReviewSummary } from './review-summary';

export interface WorkspaceFileReviewState {
  workspaceId: string;
  path: string;
  status: 'reviewed' | 'unreviewed' | string;
  reviewedAt?: string | null;
  reviewedBy: string;
  notes?: string | null;
}

export interface ReviewCockpitFile {
  file: WorkspaceChangedFile;
  review?: WorkspaceFileReviewState | null;
}

export interface WorkspacePrComment {
  workspaceId: string;
  provider: string;
  commentId: string;
  author: string;
  body: string;
  path?: string | null;
  line?: number | null;
  url?: string | null;
  state: string;
  createdAt?: string | null;
  resolvedAt?: string | null;
  commentNodeId?: string | null;
  threadId?: string | null;
  reviewId?: number | null;
  threadResolved?: boolean;
  threadOutdated?: boolean;
  threadResolvable?: boolean;
}

export interface WorkspaceReviewCockpit {
  workspaceId: string;
  files: ReviewCockpitFile[];
  selectedDiff?: WorkspaceFileDiff | null;
  reviewSummary?: WorkspaceReviewSummary | null;
  mergeReadiness?: WorkspaceMergeReadiness | null;
  prComments: WorkspacePrComment[];
  warnings: string[];
}

export interface MarkWorkspaceFileReviewedInput {
  workspaceId: string;
  path: string;
  reviewed: boolean;
  notes?: string | null;
}

export interface QueueReviewAgentPromptInput {
  workspaceId: string;
  path?: string | null;
  commentId?: string | null;
  action: 'fix_file' | 'address_comment' | 'explain_diff' | 'prepare_pr_summary' | string;
  profileId?: string | null;
  taskMode?: string | null;
  reasoning?: string | null;
  mode?: 'send_now' | 'interrupt_send' | string | null;
}

export type QueueReviewAgentPromptResult = AgentPromptEntry;

File: src-tauri/src/models/review_cockpit.rs
use serde::{Deserialize, Serialize};

use crate::models::{
    WorkspaceChangedFile, WorkspaceFileDiff, WorkspaceMergeReadiness, WorkspaceReviewSummary,
};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceFileReviewState {
    pub workspace_id: String,
    pub path: String,
    pub status: String,
    pub reviewed_at: Option<String>,
    pub reviewed_by: String,
    pub notes: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ReviewCockpitFile {
    pub file: WorkspaceChangedFile,
    pub review: Option<WorkspaceFileReviewState>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct WorkspacePrComment {
    pub workspace_id: String,
    pub provider: String,
    pub comment_id: String,
    pub author: String,
    pub body: String,
    pub path: Option<String>,
    pub line: Option<u32>,
    pub url: Option<String>,
    pub state: String,
    pub created_at: Option<String>,
    pub resolved_at: Option<String>,
    pub comment_node_id: Option<String>,
    pub thread_id: Option<String>,
    pub review_id: Option<u64>,
    pub thread_resolved: bool,
    pub thread_outdated: bool,
    pub thread_resolvable: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceReviewCockpit {
    pub workspace_id: String,
    pub files: Vec<ReviewCockpitFile>,
    pub selected_diff: Option<WorkspaceFileDiff>,
    pub review_summary: Option<WorkspaceReviewSummary>,
    pub merge_readiness: Option<WorkspaceMergeReadiness>,
    pub pr_comments: Vec<WorkspacePrComment>,
    pub warnings: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MarkWorkspaceFileReviewedInput {
    pub workspace_id: String,
    pub path: String,
    pub reviewed: bool,
    pub notes: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct QueueReviewAgentPromptInput {
    pub workspace_id: String,
    pub path: Option<String>,
    pub comment_id: Option<String>,
    pub action: String,
    pub profile_id: Option<String>,
    pub task_mode: Option<String>,
    pub reasoning: Option<String>,
    pub mode: Option<String>,
}
